native/src/components/CitySelector.tsx

```diff
@@ -140,14 +140,16 @@ class CitySelector extends React.PureComponent<PropsType> {
         <CityGroupContainer>
           <CityGroup theme={theme}>{t('nearbyPlaces')}</CityGroup>
           <NearbyMessageContainer>
-            <NearbyMessage theme={theme}>{location ? t(location.message) : ''}</NearbyMessage>
+            <NearbyMessage theme={theme}>
+              {locationState.status !== 'ready' ? t(locationState.message) : ''}
+            </NearbyMessage>
             <RetryButtonContainer>
-              {retryDetermineLocation && (
+              {locationState.status === 'unavailable' && locationState.message === 'loading' && (
```

Comment:
otherwise the user should have denied the permission for a location or the location service is disabled. But we could probably show a button if there is a timeout.

Reply:
I think showing a button always makes sense. If the permission is denied it should open the app settings to change that, if the location service is disabled maybe the location service settings.

Reply:
You are right. I checked how it was before and it should be the other way around. Nice catch 👍🏼